.vscode/settings.json

```diff
@@ -1,3 +1,12 @@
 {
-    "files.insertFinalNewline": true
+    "files.insertFinalNewline": true,
+    "[python]": {
+        "editor.defaultFormatter": "ms-python.autopep8",
+        "editor.formatOnSave": true,
```

Comment:
So with the changes I have configured, it will format on save but only format the portions of the file that changed. Additionally, some of the changes you indicate occurred BEFORE we settled on `pep8` as a linting and formatting engine and standard for the project. Introducing `pep8` with your desired config won't go back and re-write those changes. 

I included : `        "editor.formatOnSaveMode": "modifications"` in the vscode settings so that it only changes sections that were modified before the save that way it reduces future PR issues where the editor goes nuts reformatting. Keep in mind, this ONLY effects VSCode, no other editor is going to honor this setting. 

Regarding:
```
    "python.linting.pycodestyleArgs": ["--config", ".pep8"],
    "python.formatting.autopep8Args": ["--config", ".pep8"],
```

These were added to tell VSCode where it should get its `pycodestyle` settings from as by default `pycodestyle` and `autopep8` look for them in `setup.cfg`. I introduced these lines to eliminate the need for a redundant config file to achieve the same purpose.

Reply:
Okay, I thought those where manual changes. Thanks for the explanation - everything sounds reasonable. Let's go with this version.